src/clr-angular/button/button-loading/loading-button.ts

```diff
@@ -5,26 +5,91 @@
  */
 
 
-import {Component} from "@angular/core";
-import {LoadingListener} from "../../utils/loading/loading-listener";
-
+import {animate, keyframes, style, transition, trigger} from "@angular/animations";
+import {Component, ElementRef, EventEmitter, Output, Renderer2} from "@angular/core";
+import {ClrLoadingState, LoadingListener} from "../../utils/loading";
 
 @Component({
     selector: "button[clrLoading]",
     template: `
-        <span class="spinner spinner-inline" *ngIf="loading"></span>
-        <ng-content></ng-content>
+        <ng-container [ngSwitch]="state">
+            <span *ngSwitchCase="buttonState.LOADING">
+                <span @spinner class="spinner spinner-inline spinner-loading"></span>
+            </span>
+            <span *ngSwitchCase="buttonState.SUCCESS">
+                <span @validated class="spinner spinner-inline spinner-loading spinner-check"></span>
```

Comment:
`spinner-check` seems a bit weird to me. Couldn't we have just extended the `.spinner` styles and created a different class or even better, used the `clr-icon` here? We would just have to align that icon correctly in the button, no? Is the `.spinner` class adding any styles to the `checkmark` icon?

Reply:
`spinner-check` leans on the styles of `.spinner` and `.spinner-inline` and just overrides the background and animation. Using `clr-icon` would require breaking away from using any spinner classes since the `animation` and `background`are used there. Changing `spinner` class to use `clr-icon` is a breaking change for static spinner markup, so that was not an option. I created a separate ticket to look into getting rid of embedded SVGs from our components in the future: https://github.com/vmware/clarity/issues/2241. Let me know if my explanation is confusing. Happy to discuss on a call further.